pyvxl/uds.py

```diff
@@ -207,13 +218,319 @@ def ecu_reset(self, reset_type, raise_error=True, **kwargs):
+            case 0x02:
+                result = self.read_dtcs_by_status_mask(*args, **kwargs)
+            case 0x03:
+                result = self.read_dtc_snapshot_identification(*args, **kwargs)
+            case 0x04:
+                result = self.read_dtc_snapshot_record(*args, **kwargs)
+            case 0x05:
+                result = self.read_dtc_data_record(*args, **kwargs)
+            case 0x06:
+                result = self.read_dtc_extended_data_record(*args, **kwargs)
+            case 0x07:
+                result = self.read_num_dtcs_by_severity_mask(*args, **kwargs)
+            case 0x08:
+                result = self.read_dtcs_by_severity_mask(*args, **kwargs)
+            case 0x09:
+                result = self.read_severity_of_dtc(*args, **kwargs)
+            case 0x0A:
+                result = self.read_all_supported_dtcs(*args, **kwargs)
+            case 0x0C:
+                result = self.read_first_confirmed_dtc(*args, **kwargs)
+            case 0x0E:
+                result = self.read_most_recent_confirmed_dtc(*args, **kwargs)
+            case _:
+                raise NotImplementedError(f'Sub-function {sub_function} is not'
+                                          ' implemented yet.')
+
+        return result
+
+    def read_dtc_count_by_status_mask(self, status_mask, raise_error=True,
+                                      **kwargs):
+        """Read the count of DTCs matching status_mask.
+
+        Service 0x19 Sub-function 01
+        """
+        result = None
+        request = self._check('status_mask', status_mask)
+        successful, data = self.send_service(0x19, request, **kwargs)
+        if not successful:
+            if raise_error:
+                raise AssertionError('Failed to Clear DTC Group '
+                                     f'0x{request[0]:02X}{request[1]:02X}')
+        else:
+            result = data[3:]  # Remove the DTC Group from the response
+        return result
+
+    def read_dtcs_by_status_mask(self, status_mask, raise_error=True,
+                                 **kwargs):
+        """Retrieve the list of DTCs that match a client defined status mask.
+
+        Service 0x19 Sub-function 02
+
+        Args:
+            status_mask (int): Status mask used to filter DTCs by status.
+            raise_error (bool, optional): Raise error on failure.
+                                          Defaults to True.
+        """
+        result = None
+        request = (self._check('sub_function', 0x02) +
+                   self._check('status_mask', status_mask))
+        successful, data = self.send_service(0x19, request, **kwargs)
+        if not successful:
+            if raise_error:
+                raise AssertionError('Error using sub-function 0x02.')
+        else:
+            # First two bytes are sub-function ID and availability mask.
+            result = data[2:]
+
+        return result
+
+    def read_dtc_snapshot_identification(self):

```

Comment:
```suggestion
    def read_dtc_snapshot_identification(self, *args, **kwargs):
```
So someone calling this reaches the error within rather than finding an issue with an argument they're passing in first.